src/app.js

```diff
@@ -1,2 +1,254 @@
 
-window.console.info('Hello. Welcome to my website.');
\ No newline at end of file
+// Import external dependencies.
+import Ractive from 'ractive';
+import BackboneAdaptor from 'ractive-adaptors-backbone';
+import Exoskeleton from 'exoskeleton';
+import _ from 'lodash';
+import template from './app.html';
+import sass from './app.sass';
+import productRow from './product-row.html';
+import productRowEdit from './product-row-edit.html';
+import { ProductCollection, ProductModel } from './product.model';
+
+// Configure our BB Adaptor
+BackboneAdaptor.Backbone = Exoskeleton;
+
+// Create our global Ractive application instance.
+const app = Ractive({
+    target: '#BrandonsFTProject',
+    template,
+    adapt: [BackboneAdaptor],
+    partials: {
+        productRow,
+        productRowEdit,
+    },
+    data: function () {
+        return {
+
+            // UI controls - These are all modified directly by the user. Our computed properties respond.
+            itemsPerPage: 10,
+            pageIdx: 0,
+            searchText: '',
+            sortColumn: 'name',
+            sortDesc: false,
+            editingAll: false,
+
+            // products - This stores ALL products from the server. The UI does not bind directly to
+            //            this property. Instead it triggers computed-properties.
+            products: new ProductCollection(),
+
+            // editing - This is a map of draft Product Models by id. When a user edits a product,
+            //           they're actually creating a cloned Product Model instance.
+            editing: {},
+
+            // new products - This is a collection of new models without IDs. This will render
+            //                editing rows at the bottom and impacts table-page generation.
+            newProducts: new ProductCollection(),
+        }
+    },
+    computed: {
+
+        // These properties are generated whenever any of the 'this.get(...)' properties change.
+        // Ractive Computed Properties - https://ractive.js.org/api/#computed
+
+        // sorted and filtered products - a vanilla array of Product Model instances. It's been 
+        //                                sorted and filtered based on sort column, sort direction, and search text.
+        sortedAndFilteredProducts: function () {
+
+            // Cache references to our watched properties.
+            const searchText = this.get('searchText').toLowerCase();
+            const sortColumn = this.get('sortColumn');
+            const sortDesc = this.get('sortDesc');
+            let products = this.get('products');
```

Comment:
... so whenever any of these view-data properties change, this property re-calculates and then triggers additional computed-properties to also update.  This keeps our UI reactive to changes in a granular, cascading way.